pennylane/operation.py

```diff
@@ -965,70 +921,42 @@ def compute_decomposition(*params, wires=None, **hyperparameters):
-            wires (Iterable): trainable parameters of this operator, as stored in ``op.wires``
-            hyperparams (dict): non-trainable hyperparameters of this operator, as stored in ``op.hyperparameters``
-
-        Returns:
-            list[.Operator]: A list of operators.
-
-        **Example**
-
-        >>> qml.PauliX.compute_diagonalizing_gates(wires="q1")
-        [Hadamard(wires=["q1"])]
-        """
-        raise DiagGatesUndefinedError
-
-    def diagonalizing_gates(self):  # pylint:disable=no-self-use
-        r"""Defines a partial representation of this operator via
-        its eigendecomposition.
-
-        Given the eigendecomposition :math:`O = U \Sigma U^{\dagger}` where
-        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
-        the sequence of diagonalizing gates implements the unitary :math:`U`.
-        In other words, the diagonalizing gates rotate the state into the eigenbasis
-        of this operator.
+    def decomposition(self):
+        r"""The decomposition of the Operator into a product of more fundamental gates.
 
-        Returns ``None`` if this operator does not define its diagonalizing gates.
+        .. math:: O = O_1 O_2 \dots O_n
 
         .. note::
-
-            By default, this method calls the static method ``compute_diagonalizing_gates``,
-            which is used by subclasses to define the actual representation.
+            By default, this method calls the static method
+            :meth:`~.operation.Operator.compute_decomposition`. Unless the
+            :meth:`~.operation.Operator.compute_decomposition` has a custom signature,
+            this method should not be overwritten.
 
         Returns:
-            list[.Operator] or None: a list of operators
+            list[Operator]: The decomposition of the Operator into lower level operations
 
-        **Example**
+        **Example:**
 
-        >>> qml.PauliX(wires="q1").diagonalizing_gates()
-        [Hadamard(wires=["q1"])]
+        >>> qml.IsingXX(1.23, wires=(0,1)).decomposition()
+        [CNOT(wires=[0, 1]), RX(1.23, wires=[0]), CNOT(wires=[0, 1])]
         """
-        return self.compute_diagonalizing_gates(
+        return self.compute_decomposition(
             *self.parameters, wires=self.wires, **self.hyperparameters
         )
 
+    # =================================
+    # Attributes interfacing with tapes
+    # =================================
+
+    def queue(self, context=qml.QueuingContext):
+        """Append the operator to the Operator queue."""
+        context.append(self)
+        return self  # so pre-constructed Observable instances can be queued and returned in a single statement
+
+    # ===========================================
+    # Attributes interfacing with differentiation
+    # ===========================================
+
     def generator(self):  # pylint: disable=no-self-use
```

Comment:
I would say that this one is more general than just differentiation, it is used in a whole host of applications/algorithms 🤔

Reply:
In fact, this one feels more like a base representation?

Reply:
Remind me, is the generator defined as `G` in :math:`e^G`, or :math:`e^{iG}`? I'll add it to representations, we should also do this in the dev guide!

Reply:
It is actually defined as `e^{iGx}`! where `x` corresponds to a single trainable parameter. So it is a representation that not all operators will have, but I still feel a representation nonetheless.

We also plan on adding `Operator.exponent`, which I believe to be the more generalized version `e^G`